backend/fms_core/template_importer/row_handlers/library_qc/library_qc.py

```diff
@@ -84,11 +85,14 @@ def process_row_inner(self, sample_container, measures, process, process_measure
         if measures['concentration_nm'] is not None and measures['concentration_uL'] is not None:
             self.errors['concentration'] = 'Concentration must be specified in either nM or ng/uL, not both.'
 
+        # TODO: if it is a pool, library size is not provided. Should we force concentration in uL?
         concentration = measures['concentration_uL']
         if concentration is None:
             concentration = measures['concentration_nm']
             if concentration is not None:
-                molecular_weight = source_sample_obj.derived_sample_not_pool.library.molecular_weight_approx
+                # TODO: for pool, we should come with a better way to computer molecular weight
+                molecular_weight = source_sample_obj.derived_samples.first().library.molecular_weight_approx
```

Comment:
This cannot be assumed. molecular_weight is related mainly to strandedness (it's already a bit of an approximation). If we mix 50-50 sample of different strandedness the approximation will simply be wrong. The best way to deal with this i think is to fetch for each derived_sample the library molecular weight, library_size and volume ratio. You call the conversion for each derived sample then you multiply each by the volume ratio and add each fractional concentration up to get the whole pool concentration in ng/uL. You can wrap this in a service to have a single call here (with source_sample_obj and concentration as params) since this will be true even for a single library (volume_ratio being 1)